checked_message.go

```diff
@@ -87,6 +112,8 @@ func (m *CheckedMessage) Write(fields ...Field) {
 	}
 
 	m.next.Write(fields...)
+	m.lvl, m.next, m.tail = invalidLevel, nil, nil
```

Comment:
Not strictly necessary, but a nice safety nonetheless. If we're going to do this, let's pull this logic out into a small `reset` method and also clear out the message and logger instance.